Task 2/Task 2.1/CustomString/Class1.cs

```diff
@@ -0,0 +1,187 @@
+            else
+            {
+                for (int i = 0; i < str1.MyString.Length; i++)
+                {
+                    if (!Equals(str1.MyString[i], str2.MyString[i])) flag = false;
+                }
+
+                return flag;
+            }
+        }
+
+        public static bool operator != (CustomString str1, CustomString str2)
+        {
+            bool flag = false;
+            if (str1.MyString.Length != str2.MyString.Length)
+            {
+                flag = true;
+                return flag;
+            }
+            else
+            {
+                for (int i = 0; i < str1.MyString.Length; i++)
+                {
+                    if (!Equals(str1.MyString[i], str2.MyString[i])) 
+                    {
+                        flag = true;
+                    }
+                }
+
+                return flag;
+            }
+        }
+
+        /// <summary>
+        /// Метод, определяющий, есть ли искомый символ в кастомной строке.
+        /// </summary>
+        /// <param name="Symbol"></param>
+        /// <returns>Возвращает true или false, в зависимости от наличия символа в строке.</returns>
+        public bool SearchSymbol(char Symbol)
+        {
+            bool flag = false;
+            foreach (char item in MyString)
+            {
+                if (item == Symbol)
+                    flag = true;
+            }
+            return flag;
+        }
+
+        /// <summary>
+        /// Метод, подсчитывающий количество искомых символов в кастомной строке.
+        /// </summary>
+        /// <param name="Symbol"></param>
+        /// <returns>Возвращает количество символов.</returns>
+        public int CountSymbol(char Symbol)
+        {
+            int count = 0;
+
+            foreach (char item in MyString)
+            {
+                if (item == Symbol) count++;
+            }
+
+            return count;
+        }
+
+        public void Print()
+        {
+            foreach (char item in MyString)
+            {
+                Console.Write(item);
+            }
+        }
+
+        /// <summary>
+        /// Метод, возвращающий строку, состоящую из символов, хранящихся в кастомной строке,
+        /// собранных в порядке справа-налево.
+        /// </summary>
+        /// <returns></returns>
+        public string Reverse()
+        {
+            string reverse = string.Empty;
+
+            for (int i = MyString.Length - 1; i >= 0; i--)
+            {
+                reverse += MyString[i];
+            }
+
+            return reverse;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as CustomString);
+        }
+
+        public bool Equals(CustomString other)
+        {
+            return other != null &&
+                   EqualityComparer<char[]>.Default.Equals(_myString, other._myString) &&
+                   EqualityComparer<char[]>.Default.Equals(MyString, other.MyString);
+        }
```

Comment:
You don't usually trust default comparers, if you are not sure that the type has equals that does what you want.

In case of arrays:
```csharp
var comparer = EqualityComparer<char[]>.Default;
comparer.Equals(new[] { 'a', 'b', 'c' }, new[] { 'a', 'b', 'c' })
> false
```

Also you don't usually trust generated Equals - here it's comparing the same array twice